### ISSUE TITLE
Daltonization incorrect for Deuteranopia and Tritanopia?

### ISSUE THREAD

**Issue by dektar** [DaltonLens/DaltonLens#12]
It looks like the Daltonization code in https://github.com/DaltonLens/DaltonLens/blob/master/Dalton/Filters.cpp#L275 is always using the Protanopia matrix to add the error back into the image. I think it should be changing the matrix for Deuteranopia to shift away from the green channel and for Tritanopia to shift away from the blue channel.

See for example the matrixes in https://source.chromium.org/chromium/chromium/src/+/main:ui/accessibility/extensions/colorenhancer/src/cvd.js;l=106?q=cvd.js showing matrixes for correcting in red, green and blue.

**Comment by nburrus** [comment]
indeed! Actually I see I had some comments about that in the OpenGL version of the filter https://github.com/DaltonLens/DaltonLens/blob/c794984cfa55ffcf4d09bcc5281de4230aaff199/Dalton/OpenGL_Shaders.cpp#L177 . Feel free to submit a PR or I'll fix it when I get a chance, thanks for the report!

**Comment by dektar** [comment]
I'll do it if I find time, meanwhile I'm doing the similar thing [here](https://chromium-review.googlesource.com/c/chromium/src/+/4304804/4/ash/color_enhancement/color_enhancement_controller.cc#116 ) so it was helpful to confirm that the error correction matrix should differ for each CVD type!
